.golangci.yml

```diff
@@ -30,5 +30,5 @@ linters-settings:
       - indexAlloc
       - deprecatedComment
   cyclop:
-    max-complexity: 15
+    max-complexity: 20
```

Comment:
I'd try to simplify instead of raising the threshold (otherwise we can disable the lint).

Often it's enough to move some blocks to their own self-contained, self-explanatory functions.

Anyway if the task is urgent we can do it in another JIRA issue.

Reply:
well I must admit I'm not a fan of this metric (cyclomatic complexity), I don't find that it really captures the real complexity. In this case, the threshold was reached just because I added:

```go
			case limitKey:
				extraArgs.WriteString(limitParam)
				extraArgs.WriteString(param)
```
:-/

Reply:
Then we can remove it if everybody else agrees

Reply:
(what I want to pinpoint is that, sometimes, fixing cyclomatic complexity results in making the code harder to read. I'm not sating we couldn't refactor here .. but , sometimes it makes us struggle for bad reasons.

Reply:
anyway I'm rewriting the parameters processing in go to fix that